myapp/app/models/task.rb

```diff
@@ -6,4 +6,6 @@ class Task < ApplicationRecord
   enum status: { not_started: 0, in_progress: 1, done: 2 }
 
   belongs_to :user
+  has_many :task_labels, dependent: :destroy
```

Comment:
TaskLabel削除がN+1になってる？

```
  TaskLabel Destroy (1.4ms)  DELETE FROM `task_labels` WHERE `task_labels`.`id` = 1
  ↳ app/controllers/tasks_controller.rb:39:in `destroy'
  TaskLabel Destroy (1.0ms)  DELETE FROM `task_labels` WHERE `task_labels`.`id` = 2
  ↳ app/controllers/tasks_controller.rb:39:in `destroy'
  TaskLabel Destroy (1.1ms)  DELETE FROM `task_labels` WHERE `task_labels`.`id` = 3
  ↳ app/controllers/tasks_controller.rb:39:in `destroy'

  Task Destroy (0.7ms)  DELETE FROM `tasks` WHERE `tasks`.`id` = 1
```

`delete_all` にすると、1SQLで削除してくれます。コールバックは実行されないのですが、現在コールバックは使われていないので `delete_all` でも良いかもです。

```suggestion
  has_many :task_labels, dependent: :delete_all
```

```
  TaskLabel Destroy (0.8ms)  DELETE FROM `task_labels` WHERE `task_labels`.`task_id` = 2
  ↳ app/controllers/tasks_controller.rb:39:in `destroy'

  Task Destroy (1.9ms)  DELETE FROM `tasks` WHERE `tasks`.`id` = 2
  ↳ app/controllers/tasks_controller.rb:39:in `destroy'
```

他のオプションについてはこちらご参考ください！
https://railsguides.jp/association_basics.html#dependent

Reply:
また、task削除時にtask_labelが削除されるspec追加もお願いします！

Reply:
こちら修正しました！ -> [77fca43](https://github.com/Fablic/training/pull/1281/commits/77fca433bf42ee176db1ee24f8f6b6a73885ecd0)